scripts/lint-grpc-sys.pl

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/perl
+
+# Imports
+use strict;
+use warnings FATAL => 'all';
+use IPC::Open3;
+use File::Which;
+
+# Stop Perl from buffering (unimportant implementation detail)
+$|++;
+
+# Find: clang-tidy, clang-tidy-[4-6].0 in Path environment variable
+my $clangTidy = which 'clang-tidy';
+for my $ver (4 .. 6) {
+    $clangTidy = which "clang-tidy-$ver.0" if !defined $clangTidy;
+}
+die "Cannot find clang-tidy!" if !defined $clangTidy;
+
+# Variable definitions (path related)
+# Put together so people can go and modify without reading the whole script
+my $grpcSysPath = './grpc-sys';
+my $grpcPath = "$grpcSysPath/grpc";
+my $clangCompletePath = "$grpcPath/.clang_complete";
+my $extraArgFlag = ' -extra-arg-before=';
+
+# Open `.clang_complete` in grpc
+open my $clangComplete, '<', $clangCompletePath or die "Failed to open $clangCompletePath: $!\n";
```

Comment:
Why is this necessary? Isn't just command `clang-tidy -x c++ -std=c++11` sufficient enough now that `.clang-tidy` is linked?

Reply:
To make sure the command line argument passed to clang-tidy is consistent with grpc's own.

Reply:
How can they become inconsistent if not reading the files explicitly?

Reply:
According to @overvenus , our command line arguments should be the same (and always the same in the future) as grpc's own. This Perl script is just simulating grpc's [Python script](https://github.com/pingcap/grpc/blob/eb2aa9cb3b47eb3d7980952844b32082c3e5e651/tools/distrib/run_clang_tidy.py) which runs clang-tidy. The Python script reads the file, so this script reads the file as well.

The only differences are:
+ This script works with clang-tidy-[4-6].0 as well as clang-tidy, while grpc's own only works with clang-tidy.
+ This script doesn't accept arguments itself, because we don't do further customization
+ This script only checks one file, while grpc's own checks for files specified in the arguments

Reply:
How can it be always the same in the future if you choose to implement a custom script? :smiley: I think it's the same as writing a command line like provided above.